catalog/healthcheck/src/main/java/org/codice/ddf/endpoints/healthcheck/HealthChecker.java

```diff
@@ -0,0 +1,270 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.endpoints.healthcheck;
+
+import com.google.common.collect.ImmutableMap;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.SourceInfoResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.SourceInfoRequestLocal;
+import ddf.catalog.source.SourceDescriptor;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.security.Subject;
+import org.apache.karaf.bundle.core.BundleInfo;
+import org.apache.karaf.bundle.core.BundleService;
+import org.apache.karaf.bundle.core.BundleState;
+import org.apache.shiro.subject.ExecutionException;
+import org.codice.ddf.security.Security;
+import org.opengis.filter.Filter;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import java.util.Arrays;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class HealthChecker {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HealthChecker.class);
+
+  private static final Map<Integer, String> BUNDLE_STATES =
+      new ImmutableMap.Builder<Integer, String>()
+          .put(Bundle.UNINSTALLED, "UNINSTALLED")
+          .put(Bundle.INSTALLED, "INSTALLED")
+          .put(Bundle.RESOLVED, "RESOLVED")
+          .put(Bundle.STARTING, "STARTING")
+          .put(Bundle.STOPPING, "STOPPING")
+          .put(Bundle.ACTIVE, "ACTIVE")
+          .build();
+
+  private final CatalogFramework catalogFramework;
+  private BundleContext context;
+  private BundleService bundleService;
+  private Security security;
+  FilterBuilder filterBuilder;
+
+  public HealthChecker(
+      CatalogFramework catalogFramework,
+      BundleContext context,
+      BundleService bundleService,
+      FilterBuilder filterBuilder,
+      Security security) {
+    this.catalogFramework = catalogFramework;
+    this.context = context;
+    this.bundleService = bundleService;
+    this.filterBuilder = filterBuilder;
+    this.security = security;
+  }
+
+  public Response handle() {
+    if (!isDDFReady()) {
+      LOGGER.warn("DDF is not ready");
+      return Response.serverError().build();
```

Comment:
✏️ I would suggest including a human readable reason the system is not healthy for all of these server error responses so it is easy to see from the endpoint what is wrong when you get an unhealthy status.